create-or-update-files.js

```diff
@@ -100,26 +88,34 @@ module.exports = function(octokit, opts) {
           });
         }
 
+        if (Array.isArray(change.filesToDelete)) {
+          for (const fileName of change.filesToDelete) {
+            treeItems.push({
```

Comment:
Do we want to discard all deletions (line 109) if one of the files could not be found, or just the deletion for the file that does not exist?

Should we check if the file exists before pushing an entry in to `treeItems`?

Reply:
That's a good design choice, and since the primary feature of this plugin is to create and update files I went this way.

The way I see it:
In a case with N deletions, the plugin will have to issue a request per file to delete, meaning N additional requests.
This will make the plugin noticeably slower.

However, discarding N deletions because of a single missing file is also unfortunate.

What do you think would be the better choice? I actually tend to prefer the option of verifying files, but specifically my use case is submitting large amounts of creations/updates and small amounts of deletions.

Reply:
If we check before pushing there will be N additional requests to check if a file exists, and would allow us to delete any files that *do* existing if `ignoreDeletionFailures` is set to true (this would replace `retryOnDeleteFailure` I think

Given the use case of a high number of creations/updates and low number of deletions (I think this is the most common one) the additional HTTP requests are worth the usability enhancement in my opinion.